### PR TITLE
WIP: fix for docker client when docker daemon is not running

### LINKED ISSUE #589: Issues after creating tethys environment
I created a tethys_test environment using the command :  `conda create -n tethys_test -c tethysplatform -c conda-forge tethys-platform`

After that I activate the tethys_test environment: `conda activate tethys_test`

Then I I did the `tethys --help` after activating my tethys_test environment. I got the following error:
Every tethys command is giving the same error.

## Error

Traceback (most recent call last):
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/site-packages/urllib3/connectionpool.py", line 677, in urlopen
    chunked=chunked,
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/site-packages/urllib3/connectionpool.py", line 392, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/http/client.py", line 1262, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/http/client.py", line 1308, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/http/client.py", line 1257, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/http/client.py", line 1028, in _send_output
    self.send(msg)
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/http/client.py", line 968, in send
    self.connect()
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/site-packages/docker/transport/unixconn.py", line 43, in connect
    sock.connect(self.unix_socket)
FileNotFoundError: [Errno 2] No such file or directory

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/site-packages/requests/adapters.py", line 449, in send
    timeout=timeout
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/site-packages/urllib3/connectionpool.py", line 727, in urlopen
    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/site-packages/urllib3/util/retry.py", line 403, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/site-packages/urllib3/packages/six.py", line 734, in reraise
    raise value.with_traceback(tb)
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/site-packages/urllib3/connectionpool.py", line 677, in urlopen
    chunked=chunked,
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/site-packages/urllib3/connectionpool.py", line 392, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/http/client.py", line 1262, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/http/client.py", line 1308, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/http/client.py", line 1257, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/http/client.py", line 1028, in _send_output
    self.send(msg)
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/http/client.py", line 968, in send
    self.connect()
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/site-packages/docker/transport/unixconn.py", line 43, in connect
    sock.connect(self.unix_socket)
urllib3.exceptions.ProtocolError: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/site-packages/docker/api/client.py", line 205, in _retrieve_server_version
    return self.version(api_version=False)["ApiVersion"]
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/site-packages/docker/api/daemon.py", line 181, in version
    return self._result(self._get(url), json=True)
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/site-packages/docker/utils/decorators.py", line 46, in inner
    return f(self, *args, **kwargs)
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/site-packages/docker/api/client.py", line 228, in _get
    return self.get(url, **self._set_request_timeout(kwargs))
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/site-packages/requests/sessions.py", line 543, in get
    return self.request('GET', url, **kwargs)
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/site-packages/requests/sessions.py", line 530, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/site-packages/requests/sessions.py", line 643, in send
    r = adapter.send(request, **kwargs)
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/site-packages/requests/adapters.py", line 498, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/deepak/conda/anaconda/envs/tethys_test/bin/tethys", line 6, in <module>
    from tethys_cli import tethys_command
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/site-packages/tethys_cli/__init__.py", line 16, in <module>
    from tethys_cli.docker_commands import add_docker_parser
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/site-packages/tethys_cli/docker_commands.py", line 59, in <module>
    class ContainerMetadata(ABC):
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/site-packages/tethys_cli/docker_commands.py", line 69, in ContainerMetadata
    _docker_client = docker.from_env()
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/site-packages/docker/client.py", line 85, in from_env
    timeout=timeout, version=version, **kwargs_from_env(**kwargs)
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/site-packages/docker/client.py", line 40, in __init__
    self.api = APIClient(*args, **kwargs)
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/site-packages/docker/api/client.py", line 188, in __init__
    self._version = self._retrieve_server_version()
  File "/home/deepak/conda/anaconda/envs/tethys_test/lib/python3.7/site-packages/docker/api/client.py", line 213, in _retrieve_server_version
    'Error while fetching server API version: {0}'.format(e)
docker.errors.DockerException: Error while fetching server API version: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))

### PR DESCRIPTION
**Pull request by sdc50**
fixes #589

### CODE REVIEW

**Comment by sdc50** [comment]
Still need to update tests

**Comment by swainn** [comment]
@sdc50 let's enable the osx testing again to verify that the fix is working as we discussed in scrum.

File: setup.cfg [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [metadata]
 name = tethys-platform
-version = 3.1.0
+version = 3.1.1
```
Comment (swainn) [inline setup.cfg:3]: This needs to be 3.2.0 for master.

**Review by swainn — CHANGES_REQUESTED** [review]

**Comment by sdc50** [comment]
replaced by #597

**Pull request closed without merging**